code/API_definitions/qos-profiles.yaml

```
@@ -461,6 +466,19 @@ components:
         status:
           $ref: '#/components/schemas/QosProfileStatusEnum'
 
+    QosProfileDeviceResponse:
+      description: |
+        Response object for QoS Profiles for a given device. A device identifier is provided for disambiguation when multiple device identifiers were provided in the request.
+      allOf:
+        - type: object
+          properties:
+            device:
+              description: Disambiguates the device to which the session applies when multiple device identifiers were provided
```

There is no "session" here. So at least the text need to be changed. 

But I would question if we really want to add the device to the QoSProfile objects. I would rather don't do the "device ambiguation" at all if there is no other option. The QoSProfile is more or less just a "product description", there shouldn't be a device property in.

```suggestion
              description: Disambiguates the device to which the session applies when multiple device identifiers were provided
```

---

There are three alternative options that could be considered:
- If a device identifier needs to be returned for disambiguation, put it as a separate field before all the profiles. This would be a breaking change.
- Limit the number of device identifiers that can be specified (when filtering by device) to one only. This would also be a breaking change.
- Don't give any feedback as to which device identifier was selected if the API consumer specifies more than one. Either specify a "default order" in which they will be used if present (e.g. phone number first, then IP address if phone number not present, and so on), or just leave the API consumer assuming that their multiple identifiers did match the same device.

---

For this API is would consider for now the third option with "just leave the API consumer assuming that their multiple identifiers did match the same device". The device will not be actually touched or changed by the API. If the API consumer then use the or a returned QoSProfile in one of the QoS APIs, they will get there the actual DeviceResponse (given that they use the same Device object also in this request).

There is even reason why it might be better to not return a DeviceResponse in this API: operators or aggregators who will just return a small set of static profiles might not even go to the actual network to resolve the device object -- with the DeviceResponse they would need to do so.